.github/workflows/ci.yml

```diff
@@ -61,12 +62,42 @@ jobs:
       - name: Generate profile plots
         run: nix-shell --run "python e2e/plot-gas-profiles.py gas_profiles.json --output auction-gas-profiles.png"
       - uses: actions/upload-artifact@v2.2.4
+        if: github.event_name == 'push'
```

Comment:
Is the reason for the conditional here to avoid the issues with implicit commits for PRs that you mentioned on Slack? If so, could we add a comment somewhere in the CI file noting this to help remind us in the future?

Reply:
Yeah, this is something worth mentioning.

The main issue is that, using the Github GraphQL API, I couldn't find a way to easily figure out the events of the jobs that produced an artifact; so when the script is fetching the artifacts it fetches the artifacts both for the `push` event and for the `pull-request` event. These might not be the same, because the `pull-request` event tests the merged version of the PR. So, the easiest way for me to solve this was to not upload artifacts on `pull-request` events.